src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -355,92 +355,170 @@ class Config:
                 data['month'] = '*'
             else:
-                data['month'] = ",".join(data['month'])
+                data['month'] = ','.join(data['month'])
 
-        if data.get("dayweek"):
+        if data.get('dayweek'):
             if len(data['dayweek']) == 7:
                 data['dayweek'] = '*'
             else:
                 data['dayweek'] = ",".join(data['dayweek'])
 
-        mode = data.get("mode")
-        rmodule = data.get("remotemodule")
-        if mode == 'module' and not rmodule:
-            verrors.add('remotemodule', 'This field is required')
+        mode = data.get('mode')
+        if not mode:
+            verrors.add(f'{schema}.mode', 'This field is required')
 
-        rpath = data.get("remotepath")
-        if mode == 'ssh' and not rpath:
-            verrors.add('remotepath', 'This field is required')
+        remote_module = data.get('remotemodule')
+        if mode == 'module' and not remote_module:
+            verrors.add(f'{schema}.remotemodule', 'This field is required')
 
-        username = data.get("user")
         if mode == 'ssh':
-            rport = data.get("remoteport")
-            if not rport:
-                verrors.add('remoteport', 'This field is required')
-  
-            try:
-                user = await self.middleware.call('user.query', [('username', '=', username)], {'get': True})
-            except IndexError:
-                verrors.add('user', 'Provided user does not exists')
+            remote_port = data.get('remoteport')
+            if not remote_port:
+                verrors.add(f'{schema}.remoteport', 'This field is required')
+
+            remote_path = data.get('remotepath')
+            if not remote_path:
+                verrors.add(f'{schema}.remotepath', 'This field is required')
+
+            search = os.path.join(user.pw_dir, '.ssh', 'id_[edr]*')
+            if not glob.glob(search):
+                verrors.add(
+                    f'{schema}.user',
+                    'In order to use rsync over SSH you need a user'
+                    ' with a private key (DSA/ECDSA/RSA) set up in home dir.'
+                )
             else:
-                search = os.path.join(user['home'], ".ssh", "id_[edr]*.*")
-                if not glob.glob(search):
+                for file in glob.glob(search):
+                    if '.pub' in file:
+                        # file holds a public key and it's permissions should be 644
+                        if os.stat(file).st_mode & 0o777 != 0o644:
+                            verrors.add(
+                                f'{schema}.user',
+                                f'Please correct permissions for public keys in user "{username}" ssh directory'
+                            )
+                    elif 'id_' in file:
+                        # file holds a private key and it's permissions should be 600
+                        if os.stat(file).st_mode & 0o777 != 0o600:
```

Comment:
400 is also a valid permission. You should just check that it is not group or world readable.